tutorials/prepare_dataset.md

```diff
@@ -0,0 +1,145 @@
+
+
+&nbsp;
+
+### Alpaca and Alpaca Libre
+
+&nbsp;
+
+**Alpaca**
+
+The Alpaca dataset consists of 52,000 instructions and demonstrations produced by OpenAI's text-davinci-003 engine. This data is used in instruction-tuning, helping improve the performance of language models to follow instructions.
+
+In its development, the creators leveraged the data generation methodology from the [Self-Instruct framework](https://github.com/yizhongw/self-instruct).
+
+The original [Alpaca](https://crfm.stanford.edu/2023/03/13/alpaca.html) dataset can be prepared as follows:
+
+```bash
+python scripts/prepare_alpaca.py \
+ --checkpoint_dir checkpoints/tiiuae/falcon-7b
+```
+
+&nbsp;
+
+**Alpaca Libre**
+
+[Alpaca Libre](https://github.com/mobarski/alpaca-libre) is a reimplementation or alternative to Alpaca using the same formatting.
+
+To use Alpaca Libre instead of the original Alpaca dataset, use the following command:
+
+
+```bash
+python scripts/prepare_alpaca.py \
+ --checkpoint_dir "checkpoints/tiiuae/falcon-7b" \
+ --data_file_url "https://raw.githubusercontent.com/mobarski/alpaca-libre/main/data/output/alpaca_libre_ok_tasks_v4.json" \
+ --data_file_name "alpaca_libre_data_cleaned_archive.json" \
+ --destination_path "data/alpaca_libre"
+```
+
+
+
+
+&nbsp;
+
+### Dolly
+
+The Dolly dataset is a publicly available collection of 15k instruction-following entries created by Databricks. It spans multiple behavioral domains, as described in the [InstructGPT paper](https://arxiv.org/abs/2203.02155) paper. These include areas like brainstorming, classification, closed QA, content creation, information retrieval, open QA, and summary generation.
+
+The usage is similar to the Alpaca dataset described above. Using Falcon 7b as an example, we can prepare the dataset as follows:
+
+```bash
+python scripts/prepare_dolly.py \
+ --checkpoint_dir "checkpoints/tiiuae/falcon-7b" \
+ --data_file_url "https://raw.githubusercontent.com/mobarski/alpaca-libre/main/data/output/alpaca_libre_ok_tasks_v4.json" \
+ --data_file_name "alpaca_libre_data_cleaned_archive.json" \
+ --destination_path "data/dolly"
+```
+
+
+&nbsp;
+
+### LIMA
+
+The LIMA dataset is a collection of 1,000 carefully curated prompts and responses, as described in the [LIMA: Less Is More for Alignment](https://arxiv.org/abs/2305.11206) paper. The dataset is sourced from three community Q&A websites: Stack Exchange, wikiHow, and the Pushshift Reddit Dataset. In addition, it also contains prompts and answers written and collected by the authors of the LIMA paper.
+
+The usage is similar to the Dolly dataset described above except that it requires an Hugging Face access token that you need to copy & paste from your Hugging Face account. Using Falcon 7b as an example, we can prepare the dataset as follows:
+
+```bash
+python scripts/prepare_lima.py \
+ --checkpoint_dir "checkpoints/tiiuae/falcon-7b" \
+ --hf_access_token "insert_your_token_here" \
+ --destination_path "data/lima"
```

Comment:
This matches the default value

Reply:
Yes, it's more to emphasize or show where the data goes.